py2-deprecation.md

```diff
@@ -0,0 +1,134 @@
+later release. Keeping in mind SaltStack’s policy of announcing ‘end of
+support’ at least 2 major releases before the intended changes takes place,
+Sodium or a later release has been chosen as a suitable deadline for Python 2
+deprecation.
+
+**Migration:** After carefully evaluating the list of Python 3 versions
+supported by different operating systems and by also considering the EOL for
+all these operating systems [1]* (as denoted in below grid) and [EOL of Python
+branches](https://devguide.python.org/#branchstatus)[2]*, SaltStack team
+proposes to support Python 3.4 from Sodium or a later release. Versions of
+SaltStack released before the cutoff release will continue to support python
+2.7 until they reach end of life.
+
+[![Grid.png](https://i.postimg.cc/V62Wh9nY/Grid.png)](https://postimg.cc/xJymJz67)
+
+### Note 1
+
+SaltStack currently supports master and minions running on different
+versions of Python such as salt master running on Python 2, minion running on
+Python 3 and vice versa. This support will not change in this transition as a
+change in Python interpreter does not mandate a change in network protocol.
+
+### Note 2
+
+The version of OpenSSL that SaltStack will support will be determined
+by the Python version supported by SaltStack. Check the [Python SSL
+docs](https://docs.python.org/3/library/ssl.html) for the versions of OpenSSL
+supported by each Python version, and check the [pyOpenSSL release
+page](https://pypi.org/project/pyOpenSSL/#history) to identify the which
+versions of Python are supported by pyOpenSSL.
+
+### Note 3
+
+A future SEP will define a schedule for ongoing depreciation of Python versions as they also reach their end-of-life.
+
+### Note 4
+
+A future SEP will define the salt-ssh Python deprecation schedule.
+
+## Alternatives
+[alternatives]: #alternatives
+
+- Support Python2 forever. Given that many of Salt's dependencies have pledged [to drop Python2 support](https://python3statement.org/) this would mean also supporting and vendoring our dependencies.
+- Drop support for Python 2.7 in **Neon** (tentative release date is August 2019).
+
+## Packaging Considerations
+[Packaging]: #Packaging-Considerations
+
+Which Python packaging solutions to use - EPEL (Extra Packages for Enterprise
+Linux)/SC (Software Collections) repository or monolithic packaging? 
+
+While the exact nature of packages delivered by SaltStack is not being laid out
+in this SEP, SaltStack team guarantees that package updates will be made
+available for smoother transition.
+
+
+# Actions
+[Actions]: #Actions
+
+SaltStack users will see the Python 2 deprecation implemented in the next few
+releases as follows: 
+
+1. Future releases of SaltStack will have a Python 2 deprecation warning.
+2. SaltStack will create packages that will allow for a smooth transition for users.
+3. In accordance with the schedule decided on in this SEP, testing for Python 2
```

Comment:
Will this be turned off for salt-ssh as well? even though if your remove python2 support for modules it will end it being able to be tested?

How will we make sure that salt-ssh still works against python2, since only the python2 tests run salt-ssh against python2 on the remote.